### PR TITLE
Convert async void test methods to async Task

### LINKED ISSUE #8406: Convert async void test methods to async Task
Convert `async void` test methods to `async Task` to avoid potentially unobserved exceptions to crash the test runner process.

### PR DESCRIPTION
**Pull request by xavierdecoster**
## Bug

Fixes: NuGet/Home#8406
Regression: No  

## Fix

Details: 
Converted `async void` test methods to `async Task`.
Whilst at it, also upgraded `xUnit` and `Moq` dependencies to latest stable releases.

### CODE REVIEW

File: build/config.props [hunk 1]
```diff
@@ -38,9 +38,9 @@
   <PropertyGroup>
     <NewtonsoftJsonVersionCore Condition="$(NewtonsoftJsonPackageVersion) == ''">9.0.1</NewtonsoftJsonVersionCore>
     <NewtonsoftJsonVersionCore Condition="$(NewtonsoftJsonPackageVersion) != ''">$(NewtonsoftJsonPackageVersion)</NewtonsoftJsonVersionCore>
-    <XunitVersion>2.3.1</XunitVersion>
+    <XunitVersion>2.4.1</XunitVersion>
```
Comment (nkolev92) [inline build/config.props:41]: Why do we need to update the xunit versions? 

Note that there's a bootstrapping step that downloads the actual runner in .nuget/packages.config, so we'd need to change that too.

**Review by nkolev92 — COMMENTED** [review]
Upgrade the test runner version we download to run our CI tests.

File: build/config.props (hunk 1 above)
Comment (zivkan) [reply build/config.props:41]: changing the version in `.nuget/packages.config` will probably need changes to `scripts/funcTests/runFuncTests.sh` and `build/build.proj`

Comment (xavierdecoster) [reply build/config.props:41]: I was wondering where the hidden dependencies to xunit version were... 😄 
I figured it doesn't harm to upgrade whilst at it.

Comment (xavierdecoster) [reply build/config.props:41]: Fixed in https://github.com/NuGet/NuGet.Client/pull/2973/commits/923427c7a5d204624c5c5aafb3fe501916765e00.

No changes were needed in `build/build.proj` AFAICT.

**Review by nkolev92 — APPROVED** [review]
Make sure everything is green before merging :)

File: build/config.props (hunk 1 above)
Comment (zivkan) [reply build/config.props:41]: ok, different file :)

https://github.com/NuGet/NuGet.Client/blob/ea1a5fb2afe5149389bea7befb7c8a076f457274/build/common.project.props#L30

**Review by zivkan — APPROVED** [review]

File: build/config.props (hunk 1 above)
Comment (xavierdecoster) [reply build/config.props:41]: Thanks :) https://github.com/NuGet/NuGet.Client/pull/2973/commits/47eedb3f076976d4653a33b92d3ea26f4959138f

**Pull request merged**